packages/server/src/course/course.controller.ts

```diff
@@ -26,9 +27,44 @@ export class CourseController {
 
   @Get(':id')
   async get(@Param('id') id: number): Promise<GetCourseResponse> {
-    return await CourseModel.findOne(id, {
-      relations: ['officeHours', 'queues', 'queues.staffList'],
+    // TODO: for all course endpoint, check if they're a student or a TA
+    const course = await CourseModel.findOne(id, {
+      relations: ['officeHours'],
     });
+
+    const now = new Date();
+    const MS_IN_MINUTE = 60000;
+
+    const officeHoursHappeningNow = course.officeHours.filter(
+      (e) =>
+        e.startTime.valueOf() - 15 * MS_IN_MINUTE < now.valueOf() &&
+        e.endTime.valueOf() + 1 * MS_IN_MINUTE > now.valueOf(),
+    );
+
+    const queues = await QueueModel.find({
+      where: {
+        courseId: id,
+      },
+      relations: ['staffList'],
+    });
+
+    const nonEmptyQueues = queues.filter((e) => e.staffList.length > 0);
+
+    const queuesHappeningNow = [];
+
+    for (const q of queues) {
```

Comment:
Do we want to be looping through the list of `officeHoursHappeningNow` here so that if a queue does not exist for one of the office hours that currently happening, we can create new queue?

Reply:
mans got bigger brains than us @dajinchu

Reply:
I thought we should do this when we scrape ical

Reply:
That totally works. In that case we can probably just iterate through the `officeHoursHappeningNow` instead of all the queues in a course, since we'll know each office hours has a particular `queue` with the same room associated with it. 

Also if we go that route we could also remove the `if (oh) {` part.

Reply:
cool pls approve